spharpy/spherical.py

```diff
@@ -1,6 +1,431 @@
 import numpy as np
 import scipy.special as special
 import spharpy.special as _special
+import logging as logger
+from pyfar import Coordinates
+from functools import lru_cache
+from spharpy.samplings.helpers import calculate_sampling_weights
+
+
+class SphHarm:
+    def __init__(self, n_max, coords, basis_type='complex', channel_convention='acn', inv_transform_type=None,
+                 normalization='n3d'):
+        r"""
+        Create a spherical harmonics class for transformation between spherical harmonics and signals.
+
+        Parameters
+        ----------
+        n_max : int,
+            Maximum spherical harmonic order
+        coords : :doc:`pf.Coordinates <pyfar:classes/pyfar.coordinates>`
+            Coordinate object with sampling points for which the basis matrix is
+            calculated
+        basis_type : str, optional
+            Type of spherical harmonic basis, either 'complex' or 'real'. The default is 'complex'.
+        channel_convention : str, optional
```

Comment:
This could be a bit more detailed and explain the channel ordering for both cases